usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -5529,12 +6535,14 @@ arc_release(arc_buf_t *buf, void *tag)
 
 		nhdr->b_l1hdr.b_buf = buf;
 		nhdr->b_l1hdr.b_bufcnt = 1;
+		if (ARC_BUF_ENCRYPTED(buf))
+			nhdr->b_crypt_hdr.b_ebufcnt = 1;
 		(void) refcount_add(&nhdr->b_l1hdr.b_refcnt, tag);
 		buf->b_hdr = nhdr;
 
 		mutex_exit(&buf->b_evict_lock);
 		(void) refcount_add_many(&arc_anon->arcs_size,
-		    arc_buf_size(buf), buf);
+		    HDR_GET_LSIZE(nhdr), buf);
```

Comment:
Looking at the history of this, I want to confirm/ask if this is correct to revert this?  The original zfs compressed send/recv port to ZoL had `HDR_GET_LSIZE(nhdr)` but that was a bug fixed by ZoL commit 5e8ff25644dfed600b4bdfb201ea6db003b4ad2c which changed it to `arc_buf_size(buf)` -- it seems possible the PR was created before the ZoL fix was integrated (causing the reversion), so wanted to find out if it was intentional.